src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs

```diff
@@ -13,9 +15,20 @@ internal class KubernetesEventGenerator : LinuxEventGenerator
         private const int MaxDetailsLength = 10000;
         private readonly Action<string> _writeEvent;
 
-        public KubernetesEventGenerator(Action<string> writeEvent = null)
+        public KubernetesEventGenerator(IOptions<ConsoleLoggingOptions> consoleLoggingOptions, Action<string> writeEvent = null)
```

Comment:
As you are touching this code, it's good to add `?` to writeEvent param - `Action<string>? writeEvent = null`.
You’re declaring that null is an acceptable value.